### PR TITLE
Added attributes for distinguishing claims

### PR DESCRIPTION
**Pull request by traiansf**
This PR comes to address an open issue for the Haskell backend the ability to distinguish between all path and one path claims: https://github.com/kframework/kore/issues/906

### CODE REVIEW

File: pending-documentation.md [hunk 1]
```diff
@@ -845,6 +845,22 @@ Note that in the case of Set and Map, one guarantee is that K1, K2, K3, and K4
...
+be used to clarify what kind of claim a rule is.
+
+In addition of being able to annotate a rule with one of them
+(if annotating with more at the same time, only one of them would be chosen),
+one can also annotate whole modules, to give a default claim type for all rules
+in that module.
+
+Additionally, the Haskell backend introduces an extra command line option
+for the K frontend, `--default-claim-type`, with possible values
+`all-path-claim` and `one-path-claim` to allow choosing a default type for all
```
Comment (ttuegel) [inline pending-documentation.md]: Could we use `--default-claim-type=all-path` and `--default-claim-type=one-path` (without `-claim` in the flag name) for consistency with the module attributes?

Comment (traiansf) [reply pending-documentation.md]: Thanks!

**Comment by dwightguth** [comment]
I'd like to see a test for the case when the attribute is added to an individual sentence because it doesn't look like that case works to me.

**Comment by traiansf** [comment]
@dwightguth please take another look.

**Review by dwightguth — APPROVED** [review]

**Comment by dwightguth** [comment]
Looks good but I'd like Tom to approve too before we merge.

File: haskell-backend/src/main/java/org/kframework/backend/haskell/HaskellKRunOptions.java [hunk 2]
```diff
@@ -16,6 +18,19 @@
     @Parameter(names="--haskell-backend-home", description="Directory where the Haskel backend source installation resides.")
     public String haskellBackendHome = System.getenv("KORE_HOME");
 
-    @Parameter(names="--all-path-reachability", description="Whether to interpret claims as All Path Reachability Claims.")
-    public boolean allPathReachability = false;
+    @Parameter(names="--default-claim-type", converter = SentenceTypeConverter.class, description="Default type for claims. Values: [all-path|one-path")
```
Comment (ttuegel) [inline haskell-backend/src/main/java/org/kframework/backend/haskell/HaskellKRunOptions.java]:
```suggestion
    @Parameter(names="--default-claim-type", converter = SentenceTypeConverter.class, description="Default type for claims. Values: [all-path|one-path=]")
```

**Review by ttuegel — APPROVED** [review]
I think I found one trivial typo, but it looks good to me!

**Pull request merged**